packages/domains-table/src/domains-table/__tests__/domains-table-row.tsx

```diff
@@ -241,3 +266,82 @@ test( 'transfer links use the unmapped domain for the site slug', async () => {
 		'/domains/manage/example.com/transfer/in/example.com'
 	);
 } );
+
+test( 'when a domain is selected, the checkbox is checked', () => {
+	const [ partialDomain, fullDomain ] = testDomain( {
+		domain: 'example.wordpress.com',
+		blog_id: 123,
+		primary_domain: false,
+		wpcom_domain: true,
+	} );
+
+	const fetchSiteDomains = jest.fn().mockResolvedValue( {
+		domains: [ fullDomain ],
+	} );
+
+	render(
+		<DomainsTableRow
+			domain={ partialDomain }
+			fetchSiteDomains={ fetchSiteDomains }
+			isAllSitesView
+			isSelected={ true }
+			onSelect={ noop }
+		/>
+	);
+
+	expect( screen.getByRole( 'checkbox' ) ).toBeChecked();
+} );
+
+test( 'when a domain is not selected, the checkbox is unchecked', () => {
+	const [ partialDomain, fullDomain ] = testDomain( {
+		domain: 'example.wordpress.com',
+		blog_id: 123,
+		primary_domain: false,
+		wpcom_domain: true,
+	} );
+
+	const fetchSiteDomains = jest.fn().mockResolvedValue( {
+		domains: [ fullDomain ],
+	} );
+
+	render(
+		<DomainsTableRow
+			domain={ partialDomain }
+			fetchSiteDomains={ fetchSiteDomains }
+			isAllSitesView
+			isSelected={ false }
+			onSelect={ noop }
+		/>
+	);
+
+	expect( screen.getByRole( 'checkbox' ) ).not.toBeChecked();
+} );
+
+test( 'when the user selects a domain, the onSelect function is triggered', () => {
+	const [ partialDomain, fullDomain ] = testDomain( {
+		domain: 'example.wordpress.com',
+		blog_id: 123,
+		primary_domain: false,
+		wpcom_domain: true,
+	} );
+
+	const fetchSiteDomains = jest.fn().mockResolvedValue( {
+		domains: [ fullDomain ],
+	} );
+
+	const handleSelect = jest.fn();
+
+	render(
+		<DomainsTableRow
+			domain={ partialDomain }
+			fetchSiteDomains={ fetchSiteDomains }
+			isAllSitesView
+			isSelected={ false }
+			onSelect={ handleSelect }
+		/>
+	);
+
+	fireEvent.click( screen.getByRole( 'checkbox' ) );
+
+	expect( handleSelect ).toHaveBeenCalledWith( partialDomain );
+} );
```

Comment:
Not sure if these tests are meaningful. They are asserting props and click handlers, which are, in my perception, implementation details. https://github.com/Automattic/wp-calypso/pull/80826/files#diff-6fe00545d2671e8020920e06f35a90b959a6a49d05ff0a107bcbb42032f5acf4, so I wonder if we should discard the table row tests.

Reply:
Even though we all use Testing Library and the "testing trophy" now, I'm personally not against having unit tests too. But in this case I think you're right because this behaviour is already well tested by more of an integration test.